metals/src/main/scala/scala/meta/internal/builds/MillDigest.scala

```diff
@@ -0,0 +1,92 @@
+package scala.meta.internal.builds
+import scala.util.control.NonFatal
+import scala.meta.tokens.Token
+import java.nio.file.Paths
+import scala.meta.io.AbsolutePath
+import scala.collection.mutable
+import java.security.MessageDigest
+import scala.meta.internal.mtags.MtagsEnrichments._
+
+object MillDigest extends Digestable {
```

Comment:
So this doesn't cover all corner cases yet, I think I might need to rethink how to make it simpler.

If anyone got any ideas let me know, but I think I might need to filter out whitespaces and then try matching.

Reply:
Should be better now.